donkeycar/parts/dgym.py

```diff
@@ -34,24 +35,46 @@ def __init__(self, sim_path, host="127.0.0.1", port=9091, headless=0, env_name="
                      'gyro': (0., 0., 0.),
                      'accel': (0., 0., 0.),
                      'vel': (0., 0., 0.)}
-        self.delay = float(delay)
+        self.delay = float(delay) / 1000
         self.record_location = record_location
         self.record_gyroaccel = record_gyroaccel
         self.record_velocity = record_velocity
         self.record_lidar = record_lidar
 
+        self.buffer = []
+
+    def delay_buffer(self, frame, info):
+        now = time.time()
+        buffer_tuple = (now, frame, info)
+        self.buffer.append(buffer_tuple)
+
+        # go through the buffer
+        num_to_remove = 0
+        for it in range(len(self.buffer)):
```

Comment:
You can now make this more pythonic by not using the random access iterator at all but just: 
```python
for buf in in self.buffer:
    if now - buf[0] >= self.delay
    ...
```